packages/@eris-boiler/common/src/modules/stores/load-map.ts

```diff
@@ -10,16 +10,26 @@ type Loadable<T> = string | T
 export abstract class LoadMap<T> extends ExtendedMap<string, T> {
 
   protected toLoad: Array<Loadable<T>> = []
+  protected reloadables: {[k: string]: any} = {}
 
   protected abstract _load (loadableObject: any): Promise<void> | void
+  protected abstract _reload (
+    oldLoadableObject: any,
+    loadableObject: any,
+  ): Promise<void> | void
```

Comment:
I don't think it's good practice to have a function that might not return a promise. I think you should make it async to guarantee that it will always be a promise.

Reply:
I think we see a performance boost in not creating promises if not needed

but really this was done this way because it's meant to be defined later, so it simply adds the flexibility of being either a promise or not - depending on what is necessary